libtiledbsoma/src/soma/managed_query.h

```diff
@@ -499,12 +499,16 @@ class ManagedQuery {
         const CurrentDomain& current_domain, bool is_read);
     void _fill_in_subarrays_if_dense_without_new_shape(bool is_read);
 
-    // TileDB array being queried.
-    std::shared_ptr<Array> array_;
+    // NB: the Array dtor REQUIRES that this context be alive, so member
```

Comment:
Thank you for leaving this comment! Very helpful for others in the future to understand this non-obvious failure.